jcloud/api.py

```diff
@@ -95,14 +95,101 @@ def cleanup(dt) -> str:
                     cleanup(k['ctime']),
                 )
             console.print(_t)
+        return _result
+
+
+@asyncify
+async def list(args):
+    await list_by_status(args.status)
 
 
 @asyncify
 async def remove(args):
     from rich import print
+    from rich.prompt import Confirm
+
+    from .helper import get_pbar
+
+    if args.flows == []:
+        print('[cyan]Please pass in flow(s) to remove. Exiting...[/cyan]')
+        return
+
+    # Case 1: remove single flow, using full progress bar.
+    if len(args.flows) == 1 and args.flows != ['all']:
+        await CloudFlow(flow_id=args.flows[0]).__aexit__()
+        print(f'Successfully removed Flow [green]{args.flows[0]}[/green].')
+        return
+
+    # Case 2: remove a list of selected flows, using simplied progress bar.
+    if len(args.flows) > 1:
+        if 'JCLOUD_NO_INTERACTIVE' not in os.environ:
+            confirmation_message_details = '\n'.join(args.flows)
+            confirm_deleting_all = Confirm.ask(
+                f'Selected flows: \n[red]{confirmation_message_details}\n\nAre you sure you want to delete above flows? [/red]'
+            )
+            if not confirm_deleting_all:
+                print('[cyan]No worries. Exiting...[/cyan]')
+                return
+
+        flow_id_list = args.flows
+
+    # Case 3: remove all ALIVE flows.
+    else:
+        if 'JCLOUD_NO_INTERACTIVE' not in os.environ:
+            confirm_deleting_all = Confirm.ask(
+                f'[red]Are you sure you want to delete ALL the ALIVE flows that belong to you?[/red]'
+            )
+            if not confirm_deleting_all:
+                print('[cyan]No worries. Exiting...[/cyan]')
+                return
+
+        _raw_list = await list_by_status(Status.ALIVE.value)
+        print('Above are the flows about to be deleted.\n')
 
-    await CloudFlow(flow_id=args.flow).__aexit__()
-    print(f'Successfully removed Flow [green]{args.flow}[/green]')
+        if 'JCLOUD_NO_INTERACTIVE' not in os.environ:
+            confirm_deleting_again = Confirm.ask(
+                '[red]Are you sure you want to delete them?[/red]'
+            )
+            if not confirm_deleting_again:
+                print('[cyan]No worries. Exiting...[/cyan]')
+                return
+
+        flow_id_list = [flow['id'].split('-')[-1] for flow in _raw_list]
+
+    num_flows_to_remove = len(flow_id_list)
+    pbar, pb_task = get_pbar(
+        '', total=num_flows_to_remove, disable='JCLOUD_NO_PROGRESSBAR' in os.environ
+    )
+
+    with pbar:
+        pbar.start_task(pb_task)
+        pbar.update(
+            pb_task,
+            description='Starting',
+            title=f'Removing {num_flows_to_remove} flows...',
+        )
+
+        coros = [_terminate_flow_simplified(flow) for flow in flow_id_list]
```

Comment:
You can probably do 
- `tasks = [asyncio.create_task(...) for i l] `
- use `for task in as_completed(tasks):` 
- `task.result()` to get the Flow ID.

Reply:
let me give it a try

Reply:
No issues. We can improve this later, if required.